[翻译完成] flutter.cn/src/development/platform-integration/macos/building.md (#1264)
cfug/flutter.cn: src/development/platform-integration/macos/building.md

switches and segmented controls,
are also appropriate for use on macOS.

> 尽管你可以选用任何样式或主题来构建 macOS 应用程序，但是你肯定更希望应用程序的界面风格和主题与 macOS 保持一致与统一。

Author: 本页面所有评论已处理，感谢～

the App Sandbox, and the Hardened Runtime
impact your distributable application.

> 在 Xcode 中开发，请参考苹果官网指南 [在分发前对 macOS 软件进行公证][documentation on notarizing macOS Applications] 或 [使用 App Store 分发应用程序][on distributing an application through the App Store]。
> 此外，您还应该阅读本文下方的 [macOS 支持章节](#entitlements-and-the-app-sandbox)，以了解授权机制、沙盒和强化版运行时如何影响可分发的应用程序。

Reviewer: 强化版运行时 -> 更严格的运行时？下同

```suggestion
在 Xcode 中打开应用后，请参考 Apple 官网指南
[在分发前对 macOS 软件进行公证][documentation on notarizing macOS Applications]
或 [使用 App Store 分发应用程序][on distributing an application through the App Store]。
除此之外，你还需要阅读 [授权和应用沙盒](#entitlements-and-the-app-sandbox) 部分，
了解授权机制、沙盒和更严格的运行时会如何影响分发的应用程序。
```

Author: https://developer.apple.com/cn/documentation/xcode/notarizing_macos_software_before_distribution/

Author: 这是苹果官方中文开发者文档的译名，感觉和苹果保持一致会好些？https://developer.apple.com/cn/documentation/xcode/notarizing_macos_software_before_distribution/

[Build and release a macOS app][] provides a more detailed
step-by-step walkthrough of releasing a Flutter app to the
App Store.

[构建和发布为 macOS 应用][Build and release a macOS app] 文档提供了将 Flutter 应用程序发布至 App Store 的详细步骤。

[distribute it through the macOS App Store]: {{site.apple-dev}}/macos/submit/
[documentation on notarizing macOS Applications]:{{site.apple-dev}}/documentation/xcode/notarizing_macos_software_before_distribution
[on distributing an application through the App Store]: https://help.apple.com/xcode/mac/current/#/dev067853c94
[Build and release a macOS app]: {{site.url}}/deployment/macos

## Entitlements and the App Sandbox

## 授权和应用沙盒

macOS builds are configured by default to be signed,
and sandboxed with App Sandbox.
This means that if you want to confer specific
capabilities or services on your macOS app,
such as the following:

默认情况下，macOS 构建已签名并使用 App Sandbox 进行沙盒化。
这意味着如果你想要在 macOS 应用中配置特定的功能或服务，例如：

* Accessing the internet

  访问网络

* Capturing movies and images from the built-in camera

  使用内置相机拍摄图片或视频

* Accessing files

  访问文件

Then you must set up specific _entitlements_ in Xcode.
The following section tells you how to do this.

> 你必须要在 Xcode 中指定 _entitlements_  授权文件。

Reviewer: 中文的强调使用 **

```suggestion
你必须要在 Xcode 中指定 **授权文件 (entitlements)**。
```

as they're necessary for the `debug` and `profile`
modes to function correctly.

> 在 `macos/Runner/*.entitlements` 文件中，管理沙盒配置及设置。

Reviewer: ```suggestion
`macos/Runner/*.entitlements` 文件管理了沙盒的相关配置。
```

> 当你编辑这些文件时，请不要删除原始的 `Runner-DebugProfile.entitlements` 异常（支持传入网络连接和 JIT），
> 因为它们是 `debug` 和 `profile` 模式正常运行所需的。

Reviewer: ```suggestion
当你编辑这些文件时，请不要删除原始的 `Runner-DebugProfile.entitlements` 
（支持传入网络连接和 JIT），因为调试和性能模式的正常运行需要它们。
```

edit the files directly. Unless you have a very specific
reason, you should always make identical changes to both files.

> 如果你习惯于通过 **Xcode capabilities（功能） UI** 管理授权文件，请注意它仅更新两个文件中的一个。

Reviewer: ```suggestion
如果你习惯于通过 **Xcode capabilities（功能）界面** 管理授权文件，请注意它仅会更新两个文件中的一个。
```

> 此外，在某些情况下，它会创建一个新的 entitlements 授权文件，并使用其切换项目的所有配置。
> 这两种情况都会导致一些问题。
> 因此我们建议你直接编辑这些文件。
> 除非真的有明确原因必要，你应该始终对这两个文件进行同步相同的修改。

Reviewer: ```suggestion
而在某些情况下，它会创建一个新的授权文件，并使用其切换项目的所有配置。
这两种情况都会导致一些问题，因此我们建议你直接编辑这些文件。
除非有明确的需求，否则你应该同步修改这两个文件。
```

`com.apple.security.network.client`,
which you must add if you make any network requests.

如果你打算在 [App Store][] 分发你的应用程序，你需要启用应用沙盒功能，

> 此时如果你需要添加某些插件或其他原生功能，则需管理你的应用授权。
> 例如，使用 [`file_chooser`][] 插件需要添加 `com.apple.security.files.user-selected.read-only` 或 `com.apple.security.files.user-selected.read-write` 授权。

Reviewer: ```suggestion
此时如果你需要添加某些插件或其他原生功能，则需编辑你的应用授权。
例如，使用 [`file_chooser`][] 插件需要添加 `com.apple.security.files.user-selected.read-only`
或 `com.apple.security.files.user-selected.read-write` 授权。
```

  entitlement to `Runner-Release.entitlements` as well,
  otherwise your application will work correctly for debug or
  profile testing, but will fail with release builds.

> 

Reviewer: {{site.xxx}} 内的上下需要新增空行即：
```diff
{{site.xxx}}
+
  内容
+
{{site.xxx.end}}

  **重点：** 默认情况下，允许传入网络请求的 `com.apple.security.network.server` 授权，

>   只在 `debug` 和 `profile` 模式下启用，

Reviewer: ```suggestion
  只在调试和性能模式下启用，
```

>   这是为了使得 Flutter tools 能和运行中的应用进行通信。

Reviewer: ```suggestion
  这是为了让 Flutter tools 能和运行中的应用进行通信。
```

>   如果你需要允许应用的传入网络请求，则必须在 `Runner-Release.entitlements` 中添加 `com.apple.security.network.server` 授权，
>   否则你的应用仅可在 `debug` 和 `profile` 模式下正常工作，但在发布版本中失败。

Reviewer: ```suggestion
  如果你需要允许应用的传入网络请求，则必须在 `Runner-Release.entitlements` 中添加
  `com.apple.security.network.server` 授权，
  否则你的应用仅可在调试和性能模式下正常工作，在发布版本中则无法正常使用。
```

{{site.alert.end}}

For more information on these topics,
see [App Sandbox][] and [Entitlements][]
on the Apple Developer site.

> 你可浏览苹果开发者官网的 [应用沙盒][App Sandbox] 和 [授权][Entitlements] 文档，以获取关于此章节的更多信息。

Reviewer: ```suggestion
你可以阅读 Apple 开发者官网的 [应用沙盒][App Sandbox] 和
[授权][Entitlements] 文档，了解关于此章节的更多信息。
```

`com.apple.security.device.audio-input` (for Hardened Runtime)
and `com.apple.security.device.microphone` (for App Sandbox).

默认情况下，授权文件支持 JIT 调试构建，

> 但是，就如应用沙盒，你可能需要管理其他授权文件。

Reviewer: ```suggestion
但是在使用应用沙盒时，你可能需要管理其他授权文件。
```

如果你同时启用应用沙盒和强化版运行时，你可能需要为同一资源添加多个授权。

> 例如，麦克风访问权限同时要求 `com.apple.security.device.audio-input` （用于强化版运行时）和 `com.apple.security.device.microphone` （用于应用沙盒）。

Reviewer: ```suggestion
例如，麦克风访问权限同时要求 `com.apple.security.device.audio-input` （用于强化版运行时）
和 `com.apple.security.device.microphone` （用于应用沙盒）。
```

For more information on this topic,
see [Hardened Runtime][] on the Apple Developer site.

> 你可浏览苹果开发者官网的 [强化版运行时][Hardened Runtime] 文档，以获取关于此章节的更多信息。

Reviewer: ```suggestion
你可以阅读 Apple 开发者官网的 [强化版运行时][Hardened Runtime] 文档，以获取关于此章节的更多信息。
```
